source/constructs/api/common/query_condition.py

```diff
@@ -22,6 +23,60 @@ class QueryCondition(BaseModel):
     group_column: Optional[str] = const.EMPTY_STR
 
 
+def query_with_func_condition(query: Query, condition: QueryCondition,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 22. By comparison, 99% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 66-77 into a separate function.